### PR TITLE
Made `Mending Tanks`

### PR DESCRIPTION
**Pull request by Kroppeb**
Putting an item with the `Mending` enchantment repairs into a XP tank will repair it.
Will use `#damage / 2 (rounded up)` XP to do so with al limit of 16 XP / module tick

### CODE REVIEW

File: gm4_mending_tanks/data/gm4/tags/functions/init_check.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "values":[
+        "liquid_tanks:init_check"
+    ]
```
Comment (misode) [inline gm4_mending_tanks/data/gm4/tags/functions/init_check.json]: Why does this datapack contain function tags that call liquid tanks? These are already part of the liquid tanks base module. Same with `pulse_check.json`.

File: gm4_mending_tanks/data/mending_tanks/functions/item_fill.mcfunction [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#@s = liquid tank with item in first slot
+#run from liquid_tanks:item_process
+
+
+#experience tank
+execute if score @s[tag=gm4_lt_experience] gm4_lt_value matches 1.. if block ~ ~ ~ hopper{CustomName:"{\"text\":\"Experience Tank\"}",Items:[{Slot:0b,tag:{Enchantments:[{id:"minecraft:mending"}]}}]} run function mending_tanks:check_item
```
Comment (misode) [inline gm4_mending_tanks/data/mending_tanks/functions/item_fill.mcfunction]: This is called from liquid tanks so it shouldn't need to check for the name of the block. That will cause problems when an update changes it (eg. using translation keys).

File: gm4_mending_tanks/data/mending_tanks/functions/init.mcfunction [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#announce module installation
+tellraw @a[gamemode=creative] ["",{"text":"[GM4]: Installing Mending Tanks..."}]
```
Comment (misode) [inline gm4_mending_tanks/data/mending_tanks/functions/init.mcfunction]: Use translation keys for these tellraws. See the [template pack](https://github.com/Gamemode4Dev/GM4_Datapacks/blob/master/gm4_template_pack/data/template/functions/init.mcfunction) as an example. Also an issue on line 16

File: gm4_mending_tanks/data/mending_tanks/functions/init.mcfunction [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+execute unless entity @p run say GM4: Installing Mending Tanks...
```
Comment (misode) [inline gm4_mending_tanks/data/mending_tanks/functions/init.mcfunction]: Use `@a[limit=1]`. Also an issue on line 17.

File: gm4_mending_tanks/data/gm4/tags/functions/init_check.json (hunk 1 above)
Comment (Kroppeb) [reply gm4_mending_tanks/data/gm4/tags/functions/init_check.json]: Should be `mending_tanks:...` I'll fix those

File: gm4_mending_tanks/data/mending_tanks/functions/item_fill.mcfunction (hunk 2 above)
Comment (Kroppeb) [reply gm4_mending_tanks/data/mending_tanks/functions/item_fill.mcfunction]: Tell that to special. If he removes those checks, I will do so too.

File: gm4_mending_tanks/data/mending_tanks/functions/init.mcfunction (hunk 3 above)
Comment (Kroppeb) [reply gm4_mending_tanks/data/mending_tanks/functions/init.mcfunction]: I just copied liquid tanks I'll fix this.

**Comment by sparks0** [comment]
Not gonna be accepting new modules for 1.13. Please submit things for the 1.14 branch. You can always make a PR for a 1.13 version afterwards if you want back support

**Pull request closed without merging**